blueprints/public.py

```diff
@@ -1,42 +1,119 @@
-from flask import Blueprint, render_template, request, send_file, make_response
+from flask import Blueprint, render_template, request
 from modules.db import db
 from modules.email import send_confirmation
 from bson.objectid import ObjectId
 from icalendar import Calendar, Event
 from datetime import datetime
+import uuid
+
 
 public = Blueprint('public', __name__, template_folder='../templates', static_folder='../static/build/')
 
-@public.route('/')
+@public.route('/', methods=['GET'])
 def public_index():
-    return render_template('home.html')
+    if request.method == "GET":
+        return render_template('home.html')
 
 @public.route('/add', methods=['POST', 'GET'])
 def add_event():
     if request.method == 'GET':
         return render_template('add.html')
+    elif request.method == 'POST':
+        magicid = uuid.uuid4()
+        eventid = ObjectId
+        event = {
+            "_id": eventid,
+            "title": request.form["title"],
+            "description": request.form["description"],
+            "location": request.form["location"],
+            "start": request.form["start"],
+            "end": request.form["end"],
+            "rrule": request.form["rrule"],
+            "last_edited": datetime.now(timezon.utc),
+            "magic": magicid,
+        }
+        db.events.insert_one(event)
+
+        link = request.base_url + "/edit?event_id=" + eventid + "&magic=" + magicid
```

Comment:
The final url should be `/edit/<event_id>?magic=3qw4rhbo8eyo` (event_id should be part of the url, not the query string).

I changed my mind from before lol.